### PR TITLE
Implement public CI for MacOS builds

### PR DESCRIPTION
**Pull request by MaksimZhukov**
In scope of this we have configured public CI for MacOS builds

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [x] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images.CI/macos/azure-pipelines/image-generation.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        -var="vcenter_server=$(vcenter_server_v2)" \
+        -var="vcenter_username=$(vcenter_username_v2)" \
+        -var="vcenter_password=$(vcenter_password_v2)" \
+        -var="vcenter_datacenter=$(vcenter_datacenter_v2)" \
+        -var="cluster_or_esxi_host=$(esxi_cluster_v2)" \
+        -var="esxi_datastore=${{ parameters.target_datastore }}" \
+        -var="output_folder=mms-output" \
+        -var="vm_username=$(vm_username)" \
+        -var="vm_password=$(vm_password)" \
+        -var="build_id=$(Build.BuildNumber)" \
```
Comment (maxim-lobanov) [inline images.CI/macos/azure-pipelines/image-generation.yml:42]: Please make sure that build number is correct one for dev.azure.com/github/virtual-environments

File: images.CI/macos/azure-pipelines/image-generation.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  - bash: |
+      packer build -on-error=abort \
+        -var="vcenter_server=$(vcenter_server_v2)" \
+        -var="vcenter_username=$(vcenter_username_v2)" \
+        -var="vcenter_password=$(vcenter_password_v2)" \
+        -var="vcenter_datacenter=$(vcenter_datacenter_v2)" \
+        -var="cluster_or_esxi_host=$(esxi_cluster_v2)" \
+        -var="esxi_datastore=${{ parameters.target_datastore }}" \
+        -var="output_folder=mms-output" \
```
Comment (maxim-lobanov) [inline images.CI/macos/azure-pipelines/image-generation.yml:39]: Let's do `mms-output` as variable

Comment (MaksimZhukov) [reply images.CI/macos/azure-pipelines/image-generation.yml]: Thanks!

File: images.CI/macos/azure-pipelines/image-generation.yml (hunk 1 above)
Comment (MaksimZhukov) [reply images.CI/macos/azure-pipelines/image-generation.yml:42]: Thanks!

File: images.CI/macos/azure-pipelines/image-generation.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
+jobs:
+- job: Image_generation
+  displayName: Image Generation (${{ parameters.image_label }})
+  timeoutInMinutes: 720
+  pool:
+    name: Mac-Cloud V2 Image Generation
```
Comment (maxim-lobanov) [inline images.CI/macos/azure-pipelines/image-generation.yml:6]: Should be `Mac-Cloud Image Generation`

File: images.CI/macos/azure-pipelines/image-generation.yml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  timeoutInMinutes: 720
+  pool:
+    name: Mac-Cloud V2 Image Generation
+  variables:
+  - group: Mac-Cloud Image Generation
+  - group: Mac-Cloud Image Generation Key Vault
+
+  steps:
+  - pwsh: |
+      $validСontributor = ${$env:CONTRIBUTOR_ALLOWLIST}.Split(",").Trim() | Where-Object { $_ -eq $env:BUILD_SOURCEVERSIONAUTHOR } `
```
Comment (maxim-lobanov) [inline images.CI/macos/azure-pipelines/image-generation.yml:13]: `$env:CONTRIBUTOR_ALLOWLIST.Split(",")`
May be it makes sense to split this line to a few ones?

**Review by maxim-lobanov — APPROVED** [review]

File: images.CI/macos/azure-pipelines/image-generation.yml (hunk 4 above)
Comment (MaksimZhukov) [reply images.CI/macos/azure-pipelines/image-generation.yml]: thanks

**Review by al-cheb — APPROVED** [review]

**Review by maxim-lobanov — APPROVED** [review]

**Pull request merged**